src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +34,334 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
+        scale.z = D_808B45DC[i].z * (actor->scale.z * 10.0f);
+
+        func_808B39E8(&position, &scale, this->unk_16C, this->unk_170);
+
+        position.x += actor->posRot.pos.x;
+        position.y += actor->pos4.y;
+        position.z += actor->posRot.pos.z;
+
+        yIntersect = func_8003CA64(&globalCtx->colCtx, &actor->floorPoly, &bgId, actor, &position, 0);
+
+        if (returnValue < yIntersect) {
+            returnValue = yIntersect;
+            this->bgId = bgId;
+        }
+    }
+    return returnValue;
+}
+
+s32 func_808B3F58(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    if (func_808B3CA0(this, globalCtx, 0)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 1)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 2)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 3)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 4)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_808B4010(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    return !func_800435D8(globalCtx, &this->dyna, this->dyna.actor.scale.x * 290.0f,
+                          this->dyna.actor.scale.x * 290.0f + 20.0f, 1.0f);
+}
+
+void func_808B4084(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808B40AC;
+    this->dyna.actor.gravity = 0.0f;
+    this->dyna.actor.velocity.x = 0.0f;
+    this->dyna.actor.velocity.y = 0.0f;
+    this->dyna.actor.velocity.z = 0.0f;
+}
+
+void func_808B40AC(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->unk_168 <= 0 && fabsf(this->dyna.unk_150) > 0.001f) {
+        if (func_808B3AAC(this, globalCtx) && !func_808B4010(this, globalCtx)) {
+            this->unk_17C = this->dyna.unk_150;
+            func_808B4178(this, globalCtx);
+        } else {
+            player->stateFlags2 &= ~0x10;
+            this->dyna.unk_150 = 0.0f;
+        }
+    } else {
+        player->stateFlags2 &= ~0x10;
+        this->dyna.unk_150 = 0.0f;
+    }
+}
+
+void func_808B4178(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808B4194;
+    this->dyna.actor.gravity = 0.0f;
+}
+
+void func_808B4194(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    f32 sign;
+    Player* player = PLAYER;
+    f32 tempUnk178;
+    s32 approxFResult;
+    Actor* actor = &this->dyna.actor;
+
+    this->unk_174 += 0.5f;
+
+    this->unk_174 = CLAMP_MAX(this->unk_174, 2.0f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3A34.s")
+    approxFResult = Math_ApproxF(&this->unk_178, 20.0f, this->unk_174);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3A40.s")
+    sign = this->unk_17C >= 0.0f ? 1.0f : -1.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3AAC.s")
+    tempUnk178 = ((f32)sign) * this->unk_178;
```

Comment:
```suggestion
    tempUnk178 = sign * this->unk_178;
```

Reply:
I removed the brackets around it but for some reason it doesn't match if I remove the cast which is weird since it's already an f32.